Compiling in C++23 mode issues a deprecation warning for `std::aligned_storage`.

First of all thanks for the awesome library!

The new LLVM/Clang 18 implemented the deprecation of `std::aligned_storage` in C++23 and therefore it warns the following when compiling a project using `immer` in C++23 mode:

```
.../immer/detail/util.hpp:40:1: error: 'aligned_storage<8, 8>' is deprecated [-Werror,-Wdeprecated-declarations]
   40 | using aligned_storage_for =
```

of course one can switch off the warning but ideally this would be avoided by stopping using `std::aligned_storage`. 

There are some ways to avoid it as written here: https://stackoverflow.com/questions/71828288

The replacement in the SO answer cites `std::byte`, which came in C++17 afaik, but if `immer` wants to support C++14 it should be ok to just use `char`, so this means the line above would become something like:

```
template<typename T>
struct aligned_storage_for {
    alignas(T) char data[sizeof(T)] = { };
};
```

I can prepare a PR if you are interested in this fix.

--- a/immer/detail/util.hpp
+++ b/immer/detail/util.hpp
@@ -36,9 +36,20 @@ const T& as_const(T& x)
     return x;
 }
 
+template<std::size_t Size, std::size_t Align>
+struct aligned_storage
+{
+    struct type {
+        alignas(Align) unsigned char data[Size] = { };
+    };
+};
+
+template<std::size_t Size, std::size_t Align>
+using aligned_storage_t = typename aligned_storage<Size, Align>::type;
+
 template <typename T>
 using aligned_storage_for =
-    typename std::aligned_storage<sizeof(T), alignof(T)>::type;
+    typename aligned_storage<sizeof(T), alignof(T)>::type;
 
 template <typename T>
 T& auto_const_cast(const T& x)

--- a/immer/detail/util.hpp
+++ b/immer/detail/util.hpp
@@ -40,7 +40,7 @@ template<std::size_t Size, std::size_t Align>
 struct aligned_storage
 {
     struct type {
-        alignas(Align) unsigned char data[Size] = { };
+        alignas(Align) unsigned char data[Size];
     };
 };
 

--- a/immer/heap/debug_size_heap.hpp
+++ b/immer/heap/debug_size_heap.hpp
@@ -39,7 +39,10 @@ struct debug_size_heap
     constexpr static auto extra_size = 8;
 #else
     constexpr static auto extra_size = sizeof(
-        std::aligned_storage_t<sizeof(std::size_t), alignof(std::max_align_t)>);
+        detail::aligned_storage_t<
+            sizeof(std::size_t), alignof(std::max_align_t)
+        >
+    );
 #endif
 
     template <typename... Tags>

--- a/immer/heap/debug_size_heap.hpp
+++ b/immer/heap/debug_size_heap.hpp
@@ -10,6 +10,7 @@
 
 #include <immer/config.hpp>
 #include <immer/heap/identity_heap.hpp>
+#include <immer/detail/util.hpp>
 
 #include <cassert>
 #include <cstddef>

